RubberduckTests/Symbols/DeclarationFinderTests.cs

```diff
@@ -1303,6 +1304,667 @@ End Sub
+@"Option Explicit
+
+Implements TestInterface
+
+Public Property Get TestInterface_Foo(Bar As Long) As Long
+End Property
+
+Public Property Let TestInterface_Foo(Bar As Long, RHS As Long)
+End Property
+";
+            var vbe = new MockVbeBuilder()
+                .ProjectBuilder("UnderTest", ProjectProtection.Unprotected)
+                .AddComponent("TestInterface", ComponentType.ClassModule, intrface, new Selection(1, 1))
+                .AddComponent("TestImplementation", ComponentType.ClassModule, implementation, new Selection(1, 1))
+                .AddProjectToVbeBuilder()
+                .Build();
+
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var declarations = state.DeclarationFinder.AllDeclarations.ToList();
+                var declaration = declarations.Single(decl => decl.DeclarationType == DeclarationType.PropertyGet && decl.IdentifierName.Equals("Foo"));
+
+                var expected = declarations.Single(decl => decl.DeclarationType == DeclarationType.PropertyGet && decl.IdentifierName.Equals("TestInterface_Foo"));
+                var actual = state.DeclarationFinder.FindInterfaceImplementationMembers(declaration).ToList();
+                var results = actual.Count;
+
+                Assert.AreEqual(1, results, "Expected {0} Declarations, received {1}", expected, results);
+                Assert.AreEqual(expected, actual.First(), "Expected {0}, resolved to {1}", expected, actual);
+            }
+        }
+
+        [Test]
+        [Category("Resolver")]
+        [Category("Interfaces")]
+        public void FindInterfaceMembersMatchesPublicVariables()
+        {
+            var intrface =
+                @"Option Explicit
+
+Public Bar As String
+
+Public Property Get Foo() As Long
+End Property
+
+Public Property Let Foo(rhs As Long)
+End Property
+";
+
+            var implementation =
+                @"Option Explicit
+
+Implements TestInterface
+
+Public Property Get TestInterface_Bar() As String
+End Property
+
+Public Property Let TestInterface_Bar(rhs As String)
+End Property
+
+Public Property Get TestInterface_Foo() As Long
+End Property
+
+Public Property Let TestInterface_Foo(rhs As Long)
+End Property
+";
+            var vbe = new MockVbeBuilder()
+                .ProjectBuilder("UnderTest", ProjectProtection.Unprotected)
+                .AddComponent("TestInterface", ComponentType.ClassModule, intrface, new Selection(1, 1))
+                .AddComponent("TestImplementation", ComponentType.ClassModule, implementation, new Selection(1, 1))
+                .AddProjectToVbeBuilder()
+                .Build();
+
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var actual = state.DeclarationFinder.FindAllInterfaceMembers().Count();
+                const int expected = 3;
+
+                Assert.AreEqual(expected, actual, "Expected {0} Declarations, received {1}", expected, actual);
```

Comment:
The test says matches public variable but we're just counting 3 declarations without verifying they're actually the same data type? Shouldn't we assert that using `Assert.That  .... Is.Equivalent`?

Reply:
I suppose I *could*, but the count tests that all of the declared members are interface members. Maybe it just needs a better name?